server/broadcast_test.go

```diff
@@ -435,21 +588,18 @@ func TestRefreshSessions(t *testing.T) {
 	// check exit errors from createSession. Run this under -race
 	bsm = newSessionsManagerLIFO(StubBroadcastSessionsManager()) // reset bsm from previous tests
 
-	bsm.createSessions = func() ([]*BroadcastSession, error) {
-		time.Sleep(time.Millisecond)
-		return nil, fmt.Errorf("err")
-	}
+	pool := bsm.node.OrchestratorPool
+	bsm.node.OrchestratorPool = nil
 	for i := 0; i < 200; i++ {
 		wg.Add(1)
 		go func() { bsm.refreshSessions(); wg.Done() }()
 	}
 	assert.True(wgWait(&wg), "Session refresh timed out")
+	bsm.node.OrchestratorPool = pool
 
 	// check empty returns from createSession. Run this under -race
-	bsm.createSessions = func() ([]*BroadcastSession, error) {
-		time.Sleep(time.Millisecond)
-		return []*BroadcastSession{}, nil
-	}
+	updateBsmSessList(bsm.BroadcastSessionsManager, []*BroadcastSession{})
```

Comment:
I think this is different from the old test behavior. The old test behavior simulated an empty session list being returned from `createSessions()`. On L598 we assign set the orchestrator pool to an existing instance of `stubDiscovery` which already has a populated `infos` slice. This call to `updateBsmSessList()` accepts an empty session list, but the underlying `stubDiscovery` instance would still have a non-empty session list.

Reply:
fixed in 2c78fb9a784ea81c42a9bdbd0d91c71ab6fc255b